### PR TITLE
Create and test Rocky9/Rhel9/Oel9 environments for Greenplum7

### PR DESCRIPTION
**Pull request by Shakarada**
[GPR-1348,GPR-1344,GPR-1345]

## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

**Review by BaiShaoqi — COMMENTED** [review]
Since the aim is only test client, so it does not need to run the ICW group , but should have release candidate for rocky9 gpdb client binary.
another thing might need confirm is what should the job looks llike ?
Compile on rocky9 and test on oel9/rhel9/rocky9, or compile on oel9/rhel9/rocky9 and test on oel9/rhel9/rocky9 ? 
Should these jobs in another new pipeline or just stay in current rocky8 compile group https://prod.ci.gpdb.pivotal.io/teams/main/pipelines/gpdb_main?group=Compile ?

**Comment by Shakarada** [comment]
> oel9

File: concourse/pipelines/templates/gpdb-tpl.yml [hunk 1]
```diff
@@ -606,7 +612,7 @@ jobs:
       {% endif %}
 {% endif %}
 
-- name: compile_gpdb_[[ default_os_type ]]
+- name: compile_gpdb_[[ compile_platform ]]
   plan:
```
Comment (50wu) [inline concourse/pipelines/templates/gpdb-tpl.yml:616]: Previously, For RHEL, Rocky and OEL, they are all compiled on Rocky and tested on each platform. With the current change, they will be compiled and tested on each platform but we only release the Rocky binary, so I think it is not our purpose. Can e keep compiling on Rocky8 and Rocky9 and test separately?

File: concourse/pipelines/templates/gpdb-tpl.yml [hunk 2]
```diff
@@ -618,10 +624,12 @@ jobs:
           - gate_compile_start
           {% endif %}
         trigger: true
-      - get: gpdb7-[[ default_os_type ]]-build
+      - get: gpdb7-[[ compile_platform ]]-build
+      {% if compile_platform == rocky8 %}
       - get: llvm-with-asserts-packages
+      {% endif %}
   - task: compile_gpdb
```
Comment (50wu) [inline concourse/pipelines/templates/gpdb-tpl.yml:631]: The condition here is not correct, we need to test Rocky, RHEL and OEL

```
{% if compile_platform == rocky8 %}
 - get: llvm-with-asserts-packages
{% endif %}
```

File: concourse/pipelines/templates/gpdb-tpl.yml [hunk 3]
```diff
@@ -1722,13 +1758,13 @@ jobs:
                 cp -v bin_gpdb/bin_gpdb.tar.gz release_candidates/server-rc-${gpdb_semver}-el8_x86_64${RC_BUILD_TYPE_GCS}.tar.gz
                 cp -v bin_gpdb_clients/bin_gpdb_clients.tar.gz release_candidates/clients-rc-${gpdb_semver}-el8_x86_64${RC_BUILD_TYPE_GCS}.tar.gz
       - task: verify_gpdb_versions
-        image: gpdb7-[[ default_os_type ]]-build
+        image: gpdb7-[[ compile_platform ]]-build
```
Comment (50wu) [inline concourse/pipelines/templates/gpdb-tpl.yml:1761]: The release candidate platform need one condition, I still see it output is el8 actually it is for el9

```
gs://pivotal-gpdb-concourse-resources-dev/clients/published/main/clients-rc-7.0.0-alpha.0+dev.18591.gc5a654a556-el8_x86_64.debug.tar.gz
--


<pre class="timestamped-logs" style="font-family: Inconsolata, monospace; letter-spacing: 0.0425em; overflow-wrap: break-word; white-space: break-spaces; margin: 0px;"><tr class="timestamped-line" id="646ea92e:1" style="display: flex; flex-direction: row;"><a href="https://dev.ci.gpdb.pivotal.io/teams/main/pipelines/gpdb-dev-el9-test-gpdb-rocky9/jobs/Release_Candidate/builds/2#L646ea92e:1" style="color: rgb(226, 226, 226); text-decoration: none; font-family: Inconsolata, monospace; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 700; letter-spacing: 0.595px; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: break-spaces; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(54, 54, 54);"><td class="timestamp" style="padding-right: 20px; flex-shrink: 0; user-select: none; color: rgb(122, 115, 115); display: inline-block;">20:24:29</td></a><td class="timestamped-content" style="word-break: break-all; color: rgb(226, 226, 226); font-family: Inconsolata, monospace; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 700; letter-spacing: 0.595px; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: break-spaces; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(54, 54, 54); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><br class="Apple-interchange-newline">
```

**Review by 50wu — APPROVED** [review]
+1

**Pull request merged**